### DISCUSSION TITLE
How do I get username of someone who clicked a button?

### DISCUSSION THREAD

**Discussion by Sembianza** [telegraf/telegraf#1427] [Q&A]
How do I get username of someone who clicked a button?
Here's all the coding I did so far.

`bot.command(prefix + 't', (ctx) =>{
  ctx.telegram.sendMessage('@chatusername','test', {
    reply_markup: {
      inline_keyboard: [
        [{text: 'GET', callback_data: 'get'}]
      ]
    }
  })
})

bot.action('get', (ctx) =>{
  // let user = ??
  ctx.telegram.sendMessage(-1001298909547,`${user} took the thing`)
})`

**Comment by DiniFarb** [comment]
You can find that in the `ctx` object:

As in your example:

`bot.action('get', (ctx) =>{
    let user = ctx.update.callback_query.from;
    ctx.telegram.sendMessage(7777298909547,'${user.username} took the thing')
    });`

In the `from` object you'll get:
`{
  id: <int>,
  is_bot: false,
  first_name: 'name',
  username: 'only if set in Telegram',
  language_code: 'en'
}`

Please note that the property `username` is only defined if the user has set his `@username` in Telegram. Otherwise you could use the `first_name` property.

**Reply by Sembianza** [reply]
thank you so much! <3
